pkg/registry/batch/job/storage/storage_test.go

```diff
@@ -140,6 +182,178 @@ func TestDelete(t *testing.T) {
 	test.TestDelete(validNewJob())
 }
 
+type dummyRecorder struct {
+	agent string
+	text  string
+}
+
+func (r *dummyRecorder) AddWarning(agent, text string) {
+	r.agent = agent
+	r.text = text
+	return
+}
+
+func (r *dummyRecorder) getWarning() string {
+	return r.text
+}
+
+var _ warning.Recorder = &dummyRecorder{}
+
+func TestJobDeletion(t *testing.T) {
+	orphanDependents := true
+	orphanDeletionPropagation := metav1.DeletePropagationOrphan
+	backgroundDeletionPropagation := metav1.DeletePropagationBackground
+	job := validNewV1Job()
+	ctx := genericapirequest.NewDefaultContext()
+	key := "/jobs/" + metav1.NamespaceDefault + "/foo"
+	tests := []struct {
+		description   string
+		expectWarning bool
+		deleteOptions *metav1.DeleteOptions
+		requestInfo   *genericapirequest.RequestInfo
+	}{
+		{
+			description:   "DeleteOptions without any policy: warnings expected for batch/v1",
```

Comment:
nit: these are really verbose for subtest names (they tend to blow out testgrid views of runs)

suggest something like `no policy, v1, warning`, `no policy, v2, no warning`, etc